lib/src/signed_video_authenticity.c

```diff
@@ -289,25 +290,20 @@ signed_video_get_authenticity_report(signed_video_t *self)
   // Return a nullptr if no local authenticity report exists.
   if (self->authenticity == NULL) return NULL;
 
-  char *validation_str = NULL;
   signed_video_authenticity_t *authenticity_report = signed_video_authenticity_report_create();
 
   svi_rc status = SVI_UNKNOWN;
   SVI_TRY()
     SVI_THROW_IF(!authenticity_report, SVI_MEMORY);
-    validation_str = h26x_nalu_list_get_validation_str(self->nalu_list);
-    SVI_THROW(
-        allocate_memory_and_copy_string(&self->latest_validation->validation_str, validation_str));
+    // Update |number_of_pending_nalus| since that may have changed since |latest_validation|.
+    signed_video_accumulated_validation_t *accumulated = self->accumulated_validation;
+    if (accumulated->authenticity == SV_AUTH_RESULT_NOT_SIGNED) {
+      accumulated->number_of_pending_nalus = accumulated->number_of_received_nalus;
+    } else {
+      accumulated->number_of_pending_nalus = self->nalu_list->num_items;
```

Comment:
num_items comment in the struct says "The number of items linked together in the list", this doesn't accurately represent the number of pending items? Shouldn't the function h26x_nalu_list_num_pending_items be used? Haven't looked into the logic though..

Reply:
Very valid comment, and I should make this more clear in the code.
After discussions with a file player using this accumulated validation `number_of_pending_nalus` should mark the dangling end, that is, number of NALUs present after the last fully validated NALU.
This may include validated NALUs like a late arrived SEI and "ignoored" NALUs like AUD.

Maybe we should rename `number_of_pending_nalus` to `number_of_remaining_nalus` to avoid mismatch between the internal end public interpretation of 'pending'. What do you think?